### PR TITLE
Add .env support for yml file

### LINKED ISSUE #200: Feature Request: Auto-load .env file
Could we auto-load `.env` files in the root of the parent component?  This would allow users to specify environment variables in their `serverless.yml` without having to save them to their bash shell. 
 Otherwise, the user has to create custom deployment logic.

### PR DESCRIPTION
**Pull request by ebisbe**
Load environment variables with dotenv

Closes #200

* [x] Write tests
* [x] Write / update documentation
* [x] Run Prettier

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,6 +34,8 @@
     "bluebird": "^3.5.1",
     "chalk": "^2.3.2",
     "concordance": "^3.0.0",
+    "crypto-extra": "^0.4.0",
```
Comment (brianneisler) [inline package.json]: Looks like the crypto-extra dependency may have regressed. I don't think we need this dep anymore.

File: tests/integration/env/serverless.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+type: envLoad
```
Comment (brianneisler) [inline tests/integration/env/serverless.yml]: we've been trying to be consistent about naming conventions with component types. Could we rename this to `env-load` also, could we change the directory name to match the name of the component?

File: tests/integration/envLoad.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const BbPromise = require('bluebird')
+
+const fsp = BbPromise.promisifyAll(fse)
+const cpp = BbPromise.promisifyAll(cp)
+
+async function removeStateFiles(stateFiles) {
+  return BbPromise.map(stateFiles, (stateFile) => fsp.removeAsync(stateFile))
+}
+
+describe('Integration Test - Simple', () => {
```
Comment (brianneisler) [inline tests/integration/envLoad.test.js]: Looks like copy pasta. Can we update the name to match "Integration Test - Env Load". Also, can we update the name of this file to be consistent with the other tests `env-load.test.js`. Thanks!

**Comment by brianneisler** [comment]
Looks great! Thanks @ebisbe!

**Comment by ac360** [comment]
@ebisbe This is a great improvement.  Thank you for your contribution  👍

**Comment by pmuens** [comment]
Great stuff! Thanks for implementing @ebisbe 👍 💯

**Pull request merged**